logdevice/ops/ldquery/tables/Directories.cpp

```diff
@@ -0,0 +1,139 @@
+       "For logs in that directory, maximum amount of time that we can delay "
+       "delivery of newly written records.  This option increases delivery "
+       "latency but improves server and client performance."},
+      {"scd_enabled",
+       DataType::INTEGER,
+       "Indicates whether the Single Copy Delivery optimization is enabled for "
+       "this directory.  This efficiency optimization allows only one copy of "
+       "each record to be served to readers."},
+      {"custom_fields",
+       DataType::TEXT,
+       "Custom text field provided by the user."},
+  };
+}
+
+std::shared_ptr<TableData> Directories::getData(QueryContext& ctx) {
+  auto result = std::make_shared<TableData>();
+  auto full_client = ld_ctx_->getFullClient();
+
+  ld_check(full_client);
+  ClientImpl* client_impl = static_cast<ClientImpl*>(full_client.get());
+
+  auto add_row =
+      [&](const std::unique_ptr<client::Directory>& dir, const std::string dir_name, const std::string path) {
+    auto dir_attrs = dir->attrs();
+
+    ColumnValue custom;
+    if (dir_attrs.extras().hasValue() && (*dir_attrs.extras()).size() > 0) {
+      folly::dynamic extra_attrs = folly::dynamic::object;
+      for (const auto& it : dir_attrs.extras().value()) {
+        extra_attrs[it.first] = it.second;
+      }
+      custom = folly::toJson(extra_attrs);
+    }
+
+    ReplicationProperty rprop =
+        ReplicationProperty::fromLogAttributes(dir_attrs);
+    // This should remain the first ColumnValue as expected by the code below.
+    result->cols["path"].push_back(path);
+    result->cols["name"].push_back(dir_name);
+
+    result->cols["replication_property"].push_back(rprop.toString());
+    result->cols["synced_copies"].push_back(s(*dir_attrs.syncedCopies()));
+    result->cols["max_writes_in_flight"].push_back(
+        s(*dir_attrs.maxWritesInFlight()));
+    result->cols["backlog_duration_sec"].push_back(
+        s(*dir_attrs.backlogDuration()));
+    result->cols["storage_set_size"].push_back(s(*dir_attrs.nodeSetSize()));
+    result->cols["delivery_latency"].push_back(s(*dir_attrs.deliveryLatency()));
+    result->cols["scd_enabled"].push_back(s(*dir_attrs.scdEnabled()));
+    result->cols["custom_fields"].push_back(custom);
+  };
+
+  // typedef std::function<void(Status status, std::unique_ptr<client::Directory>)>
+
+  // If the query contains a constraint on the group name, we can efficiently
+  // find the directory using getLogRangeByName().
+  std::string expr;
+  if (columnHasEqualityConstraint(0, ctx, expr)) {
+      auto parent_dir = client_impl->getDirectorySync(expr);
+      if (parent_dir) {
+          const client::DirectoryMap& children = parent_dir->children();
+          for (auto& dir_pair : children) {
+              add_row(dir_pair.second, dir_pair.first, parent_dir->getFullyQualifiedName());
+          }
+      }
+  } else {
+      std::unique_ptr<client::Directory> root_dir
+          = client_impl->getDirectorySync("/");
```

Comment:
The delimiter for directories is configuration in the config. So we can't just assume `/` here would be the root. Instead, you want `client->getLogNamespaceDelimiter()`.

Reply:
will use getLogNamespaceDelimiter